### PR TITLE
Allow Coercion in Set() for numbers.

### LINKED ISSUE #1659: Set() doesn't coerce Decimal to Number and vice versa!
Should `Set(x, 1); Set(x, Float(1))` work?
Today it doesn't. Since Set(x, 1) will type `x` as `decimal`.

### PR DESCRIPTION
**Pull request by jas-valgotar**
-Fixes #1659 #1669

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs [hunk 1]
```diff
@@ -69,6 +69,12 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
 
             var arg1 = argTypes[1];
 
+            // coerce number to decimal and vice versa.
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs:72]: >// coerce number to decimal and vice versa. [](http://example.com/codeflow?start=12&length=43)

Why limit coercion to numbers only? Can we allow coercions to other types (e.g., set a GUID to a string variable)?

Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs:72]: Higher chance of storing runtime errors unknowingly for other types, so I thought initially we could start with numeric and that behavior change feels _almost_ transparent to the end user. 
I am open to expanding it to other types. What's your advice?

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs:72]: I'm leaning towards just numbers, but on the fence. 

 

I know we originally hoped dec/float was "just" ordinary coercion. But we really do have 2 concepts here:

Numeric coercion (float-->dec) is "closer" than general coercion (str-->float). 

 

general coercion is really more of "pasring" than just coercion. And practically, it has a higher chance of maker error, requires a locale, and can introduce more runtime errors that could have been caught at design time.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**